packages/playwright/src/matchers/expect.ts

```diff
@@ -322,8 +322,10 @@ class ExpectMetaInfoProxyHandler implements ProxyHandler<any> {
       };
 
       const step = testInfo._addStep(stepInfo);
+      setCurrentStep(step);
```

Comment:
You are already doing the right thing in _addStep and you call it one line above, did that logic not work? Why do you need additional currentStep treatment specifically for expect?

Reply:
It didn't, because `_addStep` relied on `stepZone` to infer the current step. But `expect` only creates a `stepZone` in some cases: https://github.com/Skn0tt/playwright/blob/4fe6927c5d754f35f01ca5b4e2705e55fad1444b/packages/playwright/src/matchers/expect.ts#L348-L350

I went with the `currentStep` approach because that's what Dima suggested instead of plumbing `step` through the `expect` calls. But looking at it again, it's so much easier to just read `exepectZone`! Did that in https://github.com/microsoft/playwright/pull/34037/commits/4c345bb01e16ae76fb7d540c6df6cf1762d9301a.